### PR TITLE
bug 1205667 - get_tidied_content can return blank

### PR DESCRIPTION
**Pull request by groovecoder**
When a $compare request is made for a large revision,
we want to skip tidy_content and return a warning to the user,
so we don't block requests on the expensive tidy operation.

### CODE REVIEW

File: kuma/wiki/models.py [hunk 1]
```diff
@@ -1735,7 +1735,10 @@ def get_tidied_content(self):
                 # already and don't need to schedule it the next time
                 # we use 3 days as a limit to try it again
                 memcache.set(tidying_scheduled_cache_key, 1, 60 * 60 * 24 * 3)
-            tidied_content, errors = tidy_content(self.content)
+            if allow_blank:
+                tidied_content = ''
```
Comment (jwhitlock) [inline kuma/wiki/models.py]: I'm worried that an empty string is a valid tidied content.  I'd rather return something non-string, like ``None`` (changing param to ``allow_null`` or ``async``), to specify that no content was returned.

File: kuma/wiki/models.py [hunk 2]
```diff
@@ -1714,7 +1714,7 @@ def get_section_content(self, section_id):
         """Convenience method to extract the content for a single section"""
         return self.document.extract_section(self.content, section_id)
 
-    def get_tidied_content(self):
+    def get_tidied_content(self, allow_blank=False):
```
Comment (jwhitlock) [inline kuma/wiki/models.py]: docstring could use update for new functionality (and to remove that trailing comma after ``field,`` :grimacing:

**Comment by jwhitlock** [comment]
Oh, and test coverage :smile:

**Comment by groovecoder** [comment]
Updated with changes and tests.

**Comment by jwhitlock** [comment]
+r was able to get the "Server is rendering the content comparison. Please refresh this page in a few minutes." message, refreshed and got it again, and on third refresh got the content comparison.

**Pull request merged**